packages/types/src/fulfillment/provider.ts

```diff
@@ -0,0 +1,102 @@
+export interface IFulfillmentProvider {
+  /**
+   * @ignore
+   *
+   * Return a unique identifier to retrieve the fulfillment plugin provider
+   */
+  getIdentifier(): string
+  /**
+   * @ignore
+   *
+   * Return the available fulfillment options for the given data.
+   */
+  getFulfillmentOptions(): Promise<Record<string, unknown>[]>
+  /**
+   * @ignore
+   *
+   * Validate the given fulfillment data.
+   */
+  validateFulfillmentData(
+    optionData: Record<string, unknown>,
+    data: Record<string, unknown>,
+    context: Record<string, unknown>
+  ): Promise<any>
+  /**
+   * @ignore
+   *
+   * Validate the given option.
+   */
+  validateOption(data: Record<string, unknown>): Promise<boolean>
+  /**
+   * @ignore
+   *
+   * Check if the provider can calculate the fulfillment price.
+   */
+  canCalculate(data: Record<string, unknown>): Promise<any>
+  /**
+   * @ignore
+   *
+   * Calculate the price for the given fulfillment option.
+   */
+  calculatePrice(
+    optionData: Record<string, unknown>,
+    data: Record<string, unknown>,
+    context: Record<string, unknown>
+  ): Promise<any>
+  /**
+   * @ignore
+   *
+   * Create a fulfillment for the given data.
+   */
+  createFulfillment(
+    data: object,
+    items: object[],
+    order: object,
```

Comment:
thought(non-blocking): I wonder if we should update the signature to expect a single object with the required properties?

This can be tackled in a follow-up PR, when we start to build out an actual provider.

Reply:
yes that was my take on that too, I ve put a todo somewhere for that